reparo/syncer/translate.go

```diff
@@ -137,3 +147,63 @@ func genColsAndArgs(row [][]byte) (cols []string, args []interface{}, err error)
 
 	return
 }
+
+// parserSchemaTableFromDDL parses ddl query to get schema and table
+// ddl like `use test; create table`
+func parserSchemaTableFromDDL(ddlQuery string) (schema, table string, err error) {
+	stmts, _, err := parser.New().Parse(ddlQuery, "", "")
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, stmt := range stmts {
+		switch node := stmt.(type) {
+		case *ast.UseStmt:
+			schema = node.DBName
+		case *ast.CreateDatabaseStmt:
+			schema = node.Name
+		case *ast.DropDatabaseStmt:
+			schema = node.Name
+		case *ast.TruncateTableStmt:
+			if len(node.Table.Schema.O) != 0 {
+				schema = node.Table.Schema.O
+			}
+			table = node.Table.Name.O
+		case *ast.CreateIndexStmt:
+			if len(node.Table.Schema.O) != 0 {
+				schema = node.Table.Schema.O
+			}
+			table = node.Table.Name.O
+		case *ast.CreateTableStmt:
+			if len(node.Table.Schema.O) != 0 {
+				schema = node.Table.Schema.O
+			}
+			table = node.Table.Name.O
+		case *ast.DropIndexStmt:
+			if len(node.Table.Schema.O) != 0 {
+				schema = node.Table.Schema.O
+			}
+			table = node.Table.Name.O
+		case *ast.AlterTableStmt:
+			if len(node.Table.Schema.O) != 0 {
+				schema = node.Table.Schema.O
+			}
+			table = node.Table.Name.O
+		case *ast.DropTableStmt:
```

Comment:
It seems that the parsings of operations that only involve 1 table are identical, maybe we can merge them into 1 case to simplify this function?

Reply:
can't merge them, because the stmt interface don't have `Table`, or can you provide some example code?

Reply:
Like this?

```go
		case *ast.DropIndexStmt, *ast.AlterTableStmt:
			if len(node.Table.Schema.O) != 0 {
				schema = node.Table.Schema.O
			}
			table = node.Table.Name.O
```

Reply:
can't do this, the interface don't have Table, will get error `type ast.StmtNode has no field or method Table`

Reply:
I see, typical Go ...